### DISCUSSION TITLE
Releasing several particles from each location

### DISCUSSION THREAD

**Discussion by LauraBachmaier** [Parcels-code/Parcels#1531] [Q&A] [answered]
Hi everyone, 

I am fairly new to using Parcels so apologies if this question is is covered somewhere already. I could not find an answer unfortunately. 

I am trying to release multiple particles (10,000) per site (I have the lat/lons for 600 sites which I am passing into the ParticleSet using lat= (vector of lats) and lon=(vector of lons)) and include a random walk component for each particle. This should mimic larval dispersal. 

I try and set size = 10,000 but this leads to the following error: AssertionError: size and positions (lon, lat, depth) don't have the same lengths. I looked through the documentation and its seems size is only a valid parameter for the from_line constructor method but I do not think this method is applicable to my scenario as I want to release particles from set positions along a coastline. 

If anyone has any advice on how I could release multiple particles for each lat /lon and introduce natural variability (random walk component) into the trajectories of each of these particles it would be much appreciated.

Many Thanks :)

**Comment by michaeldenes** [comment]
Hi @LauraBachmaier, if you could provide some code on your particle set creation that would be helpful, but in the meantime, you could try the following:

If you have a list of longitudes, latitudes, and depths of particles that you want to release, you could simply 'repeat' those lists using numpy's `np.repeat` or `np.tile` functions. For example, 
```
n_particles = 10
lons = np.linspace(-1,1, n_particles)
lats = np.linspace(-5,5, n_particles)
depths = np.zeros_like(lons)

# Repeat element wise
n_repeats = 50
lons = np.repeat(lons, n_repeats)
lats = np.repeat(lats, n_repeats)
depths = np.repeat(depths, n_repeats)

# Or repeat the array over and over
lons = np.linspace(-1,1,n_particles)
lats = np.linspace(-5,5,n_particles)
depths = np.zeros_like(lons)

lons = np.tile(lons, n_repeats)
lats = np.tile(lats, n_repeats)
depths = np.tile(depths, n_repeats)
```

Using `np.repeat` you will get arrays like `[-1,-1,-1,...]` where it repeats each element of the original array before moving onto the next, whereas using `np.tile` will tile your arrays giving you something like `[-1, -0.9, -0.8, ..., -1, -0.9, -0.8, ...]`.

Let us know if this solves your problem, cheers!

**Reply by LauraBachmaier** [reply, marked as answer]
Hi Mark, 

Thank you so much for you reply. 

A friend of mine suggested something similar using the itertools module.

```
#Load module
import itertools

#Multiply lat/lons by number of particles I want to release per site 
n= 10

lat=list(itertools.chain.from_iterable(itertools.repeat(x,n) for x in lats))
lon=list(itertools.chain.from_iterable(itertools.repeat(x,n) for x in lons))
```

This has worked great for anyone looking to do something similar.

Cheers!
